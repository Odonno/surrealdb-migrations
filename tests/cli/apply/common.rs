use anyhow::Result;
use assert_fs::TempDir;
use serial_test::serial;

use crate::helpers::*;

#[test]
#[serial]
fn apply_fails_if_both_up_and_down_args_provided() -> Result<()> {
    let temp_dir = TempDir::new()?;

    scaffold_blog_template(&temp_dir)?;

    let first_migration_name = get_first_migration_name(&temp_dir)?;

    let mut cmd = create_cmd(&temp_dir)?;

    cmd.arg("apply")
        .arg("--up")
        .arg(&first_migration_name)
        .arg("--down")
        .arg(&first_migration_name);

    cmd.assert().try_failure().and_then(|assert| {
        assert.try_stderr(
            "Error: You can\'t specify both `up` and `down` parameters at the same time\n",
        )
    })?;

    Ok(())
}
