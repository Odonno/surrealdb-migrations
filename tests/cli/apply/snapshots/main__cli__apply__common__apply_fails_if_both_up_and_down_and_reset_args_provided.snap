---
source: tests/cli/apply/common.rs
expression: stderr
---
error: the argument '--up <UP>' cannot be used with:
  --down <DOWN>
  --reset

Usage: surrealdb-migrations apply --up <UP>

For more information, try '--help'.
